backend/src/test/java/com/github/chuettenrauch/mixifyapi/integration/file/controller/FileControllerTest.java

```diff
@@ -0,0 +1,179 @@
+package com.github.chuettenrauch.mixifyapi.integration.file.controller;
+
+import com.github.chuettenrauch.mixifyapi.file.model.File;
+import com.github.chuettenrauch.mixifyapi.file.service.FileService;
+import com.github.chuettenrauch.mixifyapi.user.model.Provider;
+import com.github.chuettenrauch.mixifyapi.user.model.User;
+import com.github.chuettenrauch.mixifyapi.user.repository.UserRepository;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.mock.web.MockMultipartFile;
+import org.springframework.security.oauth2.core.user.DefaultOAuth2User;
+import org.springframework.security.oauth2.core.user.OAuth2User;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.web.servlet.MockMvc;
+
+import java.util.List;
+import java.util.Map;
+
+import static org.hamcrest.Matchers.notNullValue;
+import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.oauth2Login;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+class FileControllerTest {
+
+    @Autowired
+    private MockMvc mvc;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private FileService fileService;
+
+    @Test
+    void uploadFile_whenNotLoggedIn_returnUnauthorized() throws Exception {
+        MockMultipartFile file = new MockMultipartFile("file", "image.jpg", "image/jpeg", "some image".getBytes());
+
+        this.mvc.perform(multipart("/api/files")
+                        .file(file)
+                )
+                .andExpect(status().isUnauthorized());
+    }
+
+    @Test
+    void uploadFile_whenLoggedInButUserDoesNotExistInDB_returnUnauthorized() throws Exception {
+        MockMultipartFile file = new MockMultipartFile("file", "file.txt", "text/plain", "some image".getBytes());
+
+        this.mvc.perform(multipart("/api/files")
+                        .file(file)
+                        .with(oauth2Login())
+                )
+                .andExpect(status().isUnauthorized());
```

Comment:
What's that multipart request you're using here?

Reply:
Das musste ich nehmen, um Fake Files upzuloaden. 
Unter der Haube ist das ein POST request.